File: Performance Tax In Python/Arithmetic_Op.py
import time

# Python: Software-level Arbitrary Precision
def python_large_math():

    n1 = int("123456789")
    n2 = int("987654321")
    result = 0
    
    # Perform arithmetic
    start = time.perf_counter()
    for _ in range(10000000):
        result = n1 + n2
    print(f"Python Software-level (Int): {time.perf_counter() - start:.4f}s")

    print(f"Input N1:  {n1}")
    print(f"Input N2:  {n2}")
    print(f"Result: {result}")

python_large_math()

File: Performance Tax In Python/Arithmetic_Op.cpp
#include <iostream>
#include <string>
#include <cmath>
#include <chrono>

using namespace std;
using namespace std::chrono;

// C++: Hardware-level Fixed-size Arithmetic
int main() {
    
    long long n1 = 123456789, n2 = 987654321;
    volatile long long res = 0;
    auto start = high_resolution_clock::now();
    for(int i=0; i<10000000; ++i) {
         res = n1 + n2; 
    }
    auto end = high_resolution_clock::now();
    cout << "C++ Hardware-level: " << duration<double>(end - start).count() << "s" << endl;
    
    std::cout << "Input values: " << n1 << ", " << n2 << std::endl;
    std::cout << "Result :  " << res << std::endl;


    return 0;
}